zenoh-plugin-mqtt/src/lib.rs

```diff
@@ -134,15 +136,17 @@ async fn run(
     };
 
     // declare admin space queryable
-    let admin_keyexpr_prefix = *KE_PREFIX_ADMIN_SPACE / &zsession.zid().into_keyexpr();
-    let admin_keyexpr_expr = (&admin_keyexpr_prefix) / ke_for_sure!("**");
+    use zenoh_protocol::core::ZenohIdProto;
+    let z_id_proto: ZenohIdProto = zsession.zid().into();
+    let admin_keyexpr_prefix = *KE_PREFIX_ADMIN_SPACE / &z_id_proto.into_keyexpr();
+
+    let admin_keyexpr_expr = (&admin_keyexpr_prefix) / (ke_for_sure!("**"));
     tracing::debug!("Declare admin space on {}", admin_keyexpr_expr);
     let config2 = config.clone();
     let _admin_queryable = zsession
         .declare_queryable(admin_keyexpr_expr)
         .callback(move |query| treat_admin_query(query, &admin_keyexpr_prefix, &config2))
-        .res()
-        .await
+        .wait()
```

Comment:
wait() makes this code synchronous, it should be replaced to just .await

Reply:
Ah i see, i didnt realize it made the code Sync,